components/prism-dataweave.js

```diff
@@ -0,0 +1,47 @@
+(function (Prism) {    
+    Prism.languages.dataweave = {	
+        'url': {
+            pattern:/\b(?:[A-Za-z])+:\/\/(?:[A-Za-z0-9/:\.\-_?=&])+|\burn:(?:[A-Za-z0-9:\-_\.?=&])+/
+        },
+        'property': {
+            pattern: /(?:[A-Za-z0-9_]+#)?(?:"(?:\\.|[^\\"\r\n])*"|[A-Za-z0-9_]+)(?=\s*[:@])/,
+            greedy: true
+        },
+        'string': {
+            pattern: /(["'`])(?:\\[\s\S]|(?!\1)[^\\])*\1/,
+            greedy: true
+        },
+        'mime-type': {
+            pattern: /(?:text|audio|video|application|multipart|image)\/(?:[A-Za-z-_+0-9])+/
+        },       
+        'date': {
+            pattern: /\|[^ ](?:[^\\|\r\n]|\\[^\r\n])+\|/,
+            greedy: true
+        },
+        'comment': [
+            {
+                pattern: /(^|[^\\])\/\*[\s\S]*?(?:\*\/|$)/,
+                lookbehind: true,
+                greedy: true
+            },
+            {
+                pattern: /(^|[^\\:])\/\/.*/,
+                lookbehind: true,
+                greedy: true
+            }
+        ],
+        'regex': {
+            pattern: /\/[^ ](?:[^\\\/\r\n]|\\[^\r\n])+\//,
```

Comment:
1. Too bad `/a/` isn't a valid regex. Or is it? ;)
1. Is `/abc[/]/` not a valid regex? In languages like JS, character classes are parsed slightly differently. Is that not the case here?

```suggestion
            pattern: /\/[^ ](?:[^\\\/\r\n]|\\[^\r\n])*\//,
```

Reply:
In dataweave if you want to put a `/` you need to  escape it `\/` `/abc[/]/` needs to be `/abc[\/]/`. Regarding spaces this is wrong. Also we dataweave doesn't support empty regex `//` is not a valid regex expression so shouldn't we keep the `+`